client_test.go

```diff
@@ -102,26 +102,76 @@ func TestFindIDInConfigurationsPayload(t *testing.T) {
 	assert.Error(err)
 }
 
-func TestFindPrimaryUUID(t *testing.T) {
-	const expectedPrimary = "primary-uuid"
-	assert := assert.New(t)
-
-	var (
-		mockedHost    = "127.0.0.1"
-		mockedPrimary = expectedPrimary
-	)
+func mockGetStateNodes(assert *assert.Assertions, data []byte) *Client {
 
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		assert.Equal("GET", r.Method)
 		assert.Equal("/v1/state/nodes", r.URL.Path)
-		w.Write([]byte(fmt.Sprintf(`[{"host": "%s", "uuid": "%s"}]`, mockedHost, mockedPrimary)))
+		w.Write(data)
 	}))
 
 	host, port, err := getHostAndPortFromTestServer(ts)
 	assert.NoError(err)
 
 	c := newFlockerTestClient(host, port)
 	assert.NoError(err)
```

Comment:
Another one of these unnecessary assertions....but perhaps that's been removed in another branch.